### PR TITLE
Adds middleware locator

### PR DESCRIPTION
**Pull request by jaapio**
Based on the idea of the handlerLocator of Tactician this patch adds
a middleware locator. To make this package more independend of the other packages
of php-api-clients.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   },
   "require-dev": {
     "api-clients/test-utilities": "^1.0",
+    "container-interop/container-interop": "^1.1",
```
Comment (WyriHaximus) [inline composer.json]: Shouldn't this be in `require`?

Comment (jaapio) [inline composer.json]: Only required when using the ContainerLocator. Containers implementing this interface will always require it. So I think require-dev is sufficient.

Comment (WyriHaximus) [inline composer.json]: Yeah agree it is, but we might want to document it in the readme and add it to suggest notifying users when they install it in a project without one

Comment (jaapio) [inline composer.json]: Where do you want me to push that docs? Will add a suggestion

Comment (WyriHaximus) [inline composer.json]: The readme would be fine (which could use more docs anyway tbh).

**Comment by jaapio** [comment]
Added some basic docs should be enough to start with

**Comment by WyriHaximus** [comment]
Awesome, thanks :+1: !

**Comment by jaapio** [comment]
Cool!

**Comment by WyriHaximus** [comment]
Just tagged [`1.1.0`](https://github.com/php-api-clients/middleware/releases/tag/1.1.0) with this PR :shipit:

**Pull request merged**